### ISSUE TITLE
command_skip_current leaves an invisible MOVE command if used with queued SELFD or WAIT

### ISSUE THREAD

**Issue by salinecitrine** [beyond-all-reason/Beyond-All-Reason#2512]
### Description

command_skip_current does not properly skip an initial move command if certain other commands are queued after it. This might be an engine issue, too, since the unit continues moving without a visible MOVE order.

### Expected Behaviour

The unit should immediately stop moving, and continue to the next command (SELFD, WAIT, etc)

### Actual Behaviour

The unit receives the queued command (SELFD, WAIT, etc), but continues moving to its original destination. In addition, the move command is invisible - there is no line or command icon drawn.

### Reproduction steps

1. Issue a MOVE command to a unit
2. Issue a queued command such as SELFD or WAIT
3. execute command_skip_current ("n" in grid keys)

**Comment by sprEEEzy** [comment]
That'd be @hihoman23 's, afaik

**Comment by hihoman23** [comment]
No, I only changed behaviors for fight, patrol, and guard commands

**Comment by efrec** [comment]
I think we can move this to the engine, noting that `CMD.REMOVE` does not fully sanitize the removed commands but relies on the next command or on the end of queue.

A move goal should be a dependent of a move command, and this dependency should be cleared with the command.

**Comment by efrec** [comment]
I took another look at this. We could replace issuing `CMD.REMOVE` with `Spring.UnitFinishCommand` in a couple of places and see if it does a better job at cleaning up state.
